### ISSUE TITLE
Build Storage Persistent Volume Claim is not deleted when worker pod completes with an error

### ISSUE THREAD

**Issue by brandon-bethke-neudesic** [Azure/brigade#287]
The Persistent Volume Claim for the build shared storage is not being deleted when the brigade-worker pod completes due to an (legitimate) error.

The log for the worker is showing the error event occurring, but the 'after' event and 'beforeExit' event do not seem to be getting invoked and thus destroyStorage() is not being called. In fact the code seems to show that once the error event occurs, the process just exits, and no cleanup is performed.

See Logs
```
[brigade:k8s] default/build-images-1516919603641-776c88af phase Failed
  Error: Pod build-images-1516919603641-776c88af failed to run to completion
  
  - k8s.js:384 k.readNamespacedPod.then.response
    ./dist/k8s.js:384:32
  
[brigade:k8s] Creating secret twilio-notification-1516919607681-776c88af
[brigade:k8s] Creating pod twilio-notification-1516919607681-776c88af
[brigade:k8s] Timeout set at 900000
[brigade:k8s] default/twilio-notification-1516919607681-776c88af phase Running
[brigade:k8s] default/twilio-notification-1516919607681-776c88af phase Succeeded
[brigade:app]   Error: job build-images(build-images-1516919603641-776c88af): Error: Pod build  -images-1516919603641-776c88af failed to run to completion
  
  - brigadier.js:56 jr.run.catch.err
    ./dist/brigadier.js:56:35
  
  
  - next_tick.js:188 process._tickCallback
    internal/process/next_tick.js:188:7
  

ERROR event called { buildID: '01c4qpyrcq440h22a32xzag01e',
  workerID: 'brigade-worker-01c4qpyrcq440h22a32xzag01e-776c88af',
  type: 'error',
  provider: 'brigade',
  commit: '776c88af1336ffef1cbc3647af29c4876308154a',
  cause: 
   { event: 
      { buildID: '01c4qpyrcq440h22a32xzag01e',
        workerID: 'brigade-worker-01c4qpyrcq440h22a32xzag01e-776c88af',
        type: 'xxx',
        provider: 'xxx',
        commit: '776c88af1336ffef1cbc3647af29c4876308154a',
        payload: '' },
     reason:   Error: job build-images(build-images-1516919603641-776c88af): Error: Pod build  -images-1516919603641-776c88af failed to run to completion
  
  - brigadier.js:56 jr.run.catch.err
    ./dist/brigadier.js:56:35
  
  
  - next_tick.js:188 process._tickCallback
    internal/process/next_tick.js:188:7
  
,
     trigger: 'unhandledRejection' } }
[brigade:app] error handler is cleaning up
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.

```

**Comment by technosophos** [comment]
Okay, I'll look into that. As you note, it _should_ trigger the cleanup code right after it calls the `exit` handler that it also seems not to be calling. :-|

**Comment by brandon-bethke-neudesic** [comment]
Resolved in Pull Request https://github.com/Azure/brigade/pull/296

**Comment by technosophos** [comment]
I believe this was closed by #296's merge
